truffle/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/parser/NodeParser.java

```diff
@@ -463,6 +463,11 @@ private NodeData parseNodeData(TypeElement templateType, List<TypeElement> typeH
                 } else if (names.contains(name)) {
                     field.addError(ElementUtils.getAnnotationValue(mirror, "name"), "Duplicate field name '%s'.", name);
                 }
+
+                if (type == null) {
+                    throw new AssertionError("The type specified in the NodeField annotation cannot be found. A possible cause could be a missing import statement for the type.");
```

Comment:
This does not look like an error/invariant that should raise an AssertionError. It looks more like an error that is relevant for the user. Also, it would be nice to actually include name of the type in the error.